src/main/java/network/brightspots/rcv/CommonDataFormatReader.java

```diff
@@ -22,28 +22,100 @@
 
 package network.brightspots.rcv;
 
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
 import java.io.File;
+import java.io.FileInputStream;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.logging.Level;
 import javafx.util.Pair;
+import network.brightspots.rcv.CastVoteRecord.CvrParseException;
+import network.brightspots.rcv.RawContestConfig.CvrSource;
 import network.brightspots.rcv.TabulatorSession.UnrecognizedCandidatesException;
 
 class CommonDataFormatReader {
 
+  private final String STATUS_NEEDS_ADJUDICATION = "needs-adjudication";
+  private final String STATUS_YES = "yes";
+  private final String STATUS_NO = "no";
+  private final String STATUS_UNKNOWN = "unknown";
+  private final String BOOLEAN_TRUE = "true";
+  private final String BOOLEAN_FALSE = "false";
+
   private final String filePath;
   private final ContestConfig config;
+  private final CvrSource source;
   private Map<String, Integer> unrecognizedCandidateCounts = new HashMap<>();
 
-  CommonDataFormatReader(String filePath, ContestConfig config) {
+  CommonDataFormatReader(String filePath, ContestConfig config, CvrSource source) {
     this.filePath = filePath;
     this.config = config;
+    this.source = source;
+  }
+
+  // This method will extract candidate data from a CDF file for the contestID specified in
+  // our CvrSource.  It is currently un-used but will be handy for automating
+  // config file creation when we are ready to implement that.
+  Map<String, String> getCandidates() throws CvrParseException {
+    Map<String, String> candidates;
+    if (filePath.endsWith(".xml")) {
+      candidates = getCandidatesXml();
+    } else if (filePath.endsWith(".json")) {
+      candidates = getCandidatesJson();
+    } else {
+      Logger.log(Level.SEVERE,
+          "Unexpected file extension: %s.  CDF source files must be .xml or .json", this.filePath);
```

Comment:
I think we can and should do this check as part of the validation instead of here and in `parseCvrFile`. Recommend removing it from both and putting it in `ContestConfig` where the "// perform CDF checks" comment is.

Reply:
+1